lib/convert_query_type.js

```diff
@@ -0,0 +1,108 @@
+var ObjectId = require('mongodb').ObjectId;
+
+/**
+ * Creates an object containing the expression key and value queried.
+ *
+ * @param {String} expression - This is the operator used in query.
+ * @param {String} field - This is the column being quried.
+ * @param {ObjectId|Number|Boolean} value - This is the value being quried.
+ *
+ * @returns {Object} expressionQuery - contains the expressions key and query value.
+ */
+function query(value, field, expression) {
+    if (!expression) {
+        return value;
+    }
+
+    var expressionQuery = {};
+    expressionQuery[expression] = value;
+    return expressionQuery;
+}
+
+/**
+ * Tries to create a Number from the queried value.
+ *
+ * @param {String} value - This is the value being quried.
+ *
+ * @returns {Number}
+ */
+function number(value) {
+    var num = Number(value);
+    if (!Number.isNaN(num)) {
+        return num;
+    }
+}
+
+/**
+ * Tries to create an ObjectId from the queried value.
+ *
+ * @param {String} value - This is the value being quried.
+ *
+ * @returns {ObjectId}
+ */
+function objectId(value) {
+    try {
+        return ObjectId(value);
+    } catch (err) { }
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Either remove or fill this block of code. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AEmptyBlock)